tree/ntuple/src/RFieldBase.cxx

```diff
@@ -1028,11 +1029,74 @@ void ROOT::RFieldBase::ConnectPageSource(ROOT::Internal::RPageSource &pageSource
    for (auto &column : fAvailableColumns)
       column->ConnectPageSource(fOnDiskId, pageSource);
 
-   AfterConnectPageSource();
-
    fState = EState::kConnectedToSource;
 }
 
+void ROOT::RFieldBase::EnsureCompatibleOnDiskField(const ROOT::Internal::RPageSource &source,
+                                                   std::uint32_t ignoreBits) const
+{
+   // For computed fields added by schema evolution (e.g., new class member, new base class,
+   // field targeted by I/O customization rule), ignore on-disk information.
+   if (!fIsArtificial) {
+      const auto descriptorGuard = source.GetSharedDescriptorGuard();
+      EnsureCompatibleOnDiskField(descriptorGuard->GetFieldDescriptor(GetOnDiskId()), ignoreBits);
+   }
+}
+
+void ROOT::RFieldBase::EnsureCompatibleOnDiskField(const RFieldDescriptor &fieldDesc, std::uint32_t ignoreBits) const
```

Comment:
Unless I am mistaken, this function currently/for-now does not quite live up to his name (rather than 'compatible with' we have (for now) 'strict' equality, should we document the current state (and/or update the name if this is the long term intent)?

Reply:
It's not a strict equality. We don't require, e.g., the field description nor the type alias to be identical.

Reply:
@pcanal can this be resolved?